ingestion_server/ingestion_server/api.py

```diff
@@ -128,40 +134,53 @@ def on_get(self, req, resp):
 class TaskStatus(BaseTaskResource):
     def on_get(self, req, resp, task_id):
         """Check the status of a single task."""
-        task = self.tracker.id_task[task_id]
-        active = task.is_alive()
+        try:
+            task = self.tracker.id_task[task_id]
+        except KeyError:
+            resp.status = falcon.HTTP_404
+            resp.media = {"message": f"No task found with id {task_id}"}
+            return
```

Comment:
My be better to use `.get` rather than a try except?
```suggestion
        task = self.tracker.id_task.get(task_id)
        if task is None:
            resp.status = falcon.HTTP_404
            resp.media = {"message": f"No task found with id {task_id}"}
            return
```

Reply:
Agreed! Done